flink-runtime/src/test/java/org/apache/flink/runtime/jobmaster/TestingAbstractInvokables.java

```diff
@@ -82,4 +84,94 @@ public void invoke() throws Exception {
 			}
 		}
 	}
+
+	/**
+	 * Test invokable which blocks after invoke was called.
+	 */
+	public static class BlockingInvokable extends AbstractInvokable {
+
+		public BlockingInvokable(Environment environment) {
+			super(environment);
+		}
+
+		@Override
+		public void invoke() throws Exception {
+			final Object o = new Object();
+			//noinspection SynchronizationOnLocalVariableOrMethodParameter
+			synchronized (o) {
+				//noinspection InfiniteLoopStatement
+				while (true) {
+					o.wait();
+				}
+			}
+		}
+
+	}
+
+	/**
+	 * Test invokable use the input gate of index 0.
+	 */
+	public static class AgnosticReceiver extends AbstractInvokable{
```

Comment:
We already have `org.apache.flink.runtime.jobmanager.AgnosticReceiver`.

Reply:
@TisonKun My initial intention is to use the java version and remove the scala version of AgnosticReceiver, but to avoid mix two issues, I will change it to the scala version.